projects/vdk-plugins/vdk-notebook/src/vdk/plugin/notebook/notebook_plugin.py

```diff
@@ -1,33 +1,28 @@
 # Copyright 2021 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
-from typing import List
+from __future__ import annotations
+
+import logging
 
 from vdk.api.plugin.hook_markers import hookimpl
-from vdk.api.plugin.plugin_registry import IPluginRegistry
-from vdk.internal.core.config import ConfigurationBuilder
-from vdk.internal.core.context import CoreContext
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.plugin.notebook.notebook_based_step import JobNotebookLocator
+from vdk.plugin.notebook.notebook_reader import Notebook
+from vdk.plugin.notebook.notebook_reader import NotebookReader
 
-"""
-Include the plugins implementation. For example:
-"""
+log = logging.getLogger(__name__)
 
 
-class DummyPlugin:
-    @hookimpl(tryfirst=True)
-    def vdk_configure(self, config_builder: ConfigurationBuilder):
-        config_builder.add(
-            key="dummy_config_key",
-            default_value="dummy",
-            description="""
-                Dummy configuration
-            """,
+@hookimpl()
+def initialize_job(context: JobContext):
+    if context.job_directory is None:
+        log.info(
+            "Data Job directory is not specified. Default job initialization will be skipped."
         )
-
-    @hookimpl
-    def vdk_initialize(self, context: CoreContext):
-        print("initializing dummy")
-
-
-@hookimpl
-def vdk_start(plugin_registry: IPluginRegistry, command_line_args: List):
-    plugin_registry.load_plugin_with_hooks_impl(DummyPlugin(), "DummyPlugin")
+        return
+    file_locator: JobNotebookLocator = JobNotebookLocator()
+    notebook_files = file_locator.get_notebook_files(context.job_directory)
+    if len(notebook_files) >= 1:
+        for file_path in notebook_files:
+            nb: Notebook = Notebook(file_path)
+            NotebookReader.read_notebook_and_save_steps(nb, context)
```

Comment:
Actually, `class Notebook` definition is physically closer to `class NotebookReader`,
so why importing both to `notebook_plugin.py` to just to pass an instance of the one, to the method of the other?